README.md

```diff
@@ -0,0 +1,114 @@
+# xtm - a library for cross-thread messaging with event loop intergation
+
+The library provides the following facilities:
+
+# xtm_queue
+
+Opaque struct, that represent one-directional, one-reader-one-writer queue
+implementation with eventloop integration ability.
+
+## xtm_create
+
+Allocation function, used to create instance of struct xtm_queue and return
+its pointer or NULL with the following possible errno: ENOMEM, EINVAL, EMFILE,
+ENFILE, ENODEV. Accepts the queue size as input, which must be a power of two.
+
+## xtm_delete
+
+Deallocation function, used to free queue and close its internal fds.
+Returns 0 on success. Otherwise -1 with errno set appropriately (as in
+close(2), since it implies close of fds)
+
+## xtm_msg_notify
+
+Function for queue consumer notification.
+Returns 0 on success. Otherwise -1 with errno set appropriately (as in
+write(2), since it implies write to internal fd)
+
+## xtm_msg_probe
+
+Returns 0 if queue has space. Otherwise -1 with errno set to ENOBUFS.
+
+## xtm_msg_count
+
+Function returns current message count in queue.
+
+## xtm_fun_dispatch
+
+Function puts message containing the function and its argument to the queue.
+If delayed == 0 notifies consumer.
+Returns 0 on success. Otherwise -1 with errno set appropriately (as in
+write(2), since it implies write to internal fd + ENOBUFS in case there
+is no space in queue)
+
+## xtm_fd
+
+Returns file descriptor, that should be watched to become readable. When it
+became readable, consumer should call one of the consumer functions:
+xtm_msg_recv or xtm_fun_invoke.
+
+## xtm_fun_invoke
+
+Function retrieves messages from the queue and calls functions contained
+in them. If flushed == 1, flushing queue pipe. Return count of invoked
+functions on success. Otherwise -1 with errno set appropriately (as in
+read(2), since it implies read to internal fd)
+
+## xtm_msg_send
+
+Function puts message to the queue. If delayed == 0 notifies consumer.
+Returns 0 on success. Otherwise -1 with errno set appropriately (as in
```

Comment:
I don't like that we have 'delayed' as an option. Firstly, it adds a branch to the potentially very hot code. Secondly, when you call that function and pass 0/1 as a constant, it is super unclear what does it mean, and forces you to go to the documentation or declaration just to understand what is happening.

Reply:
I talked with @Mons about it. This is his answer:
"
This option was append special!
These are 2 different scenarios. In one, you don't think about notify and just call send. In another, you call send + delayed and then call notify. That is why this option was made. Without the option, it splits into 2 different functions.
"

Reply:
And what is the problem with having 2 different functions? I really don't see what is the issue. It would be better - easier to understand the code.

Reply:
> And what is the problem with having 2 different functions? I really don't see what is the issue. It would be better - easier to understand the code.

That's what I said.
That's what most developers voted for in Telegram.
But that's Mons...

Reply:
Even more, httpng until recently used an older version of xtm with separate functions.

Reply:
Mons is the customer, API is a part of task.
Let's leave the API as it is.
BTW we can add two specialized function later.

Reply:
This is ridiculous. Can you provide any real arguments except "Mons is the customer"?